travelpal/assets/js/components/friend.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { connect } from 'react-redux';
+import { Card, CardBody, CardHeader, Button } from 'reactstrap';
+import { Row, Col } from 'reactstrap';
+
+import api from '../api';
+
+// Renders the details of an individual friend as a card
+function Friend(props) {
+  let btnTxt = (props.status == "Accepted") ? "Unfriend" : "Cancel Request";
+
+  // Sends an unfriend request
+  function unfriend() {
+    api.delete_friend(props.id);
+  }
+
+  // Returns the friend's details as a Bootstrap card element
+  return (
+    <Col md="6">
+      <Card>
+        <CardHeader>
+          <Row>
+            <Col md="6">
+              {props.name}
+            </Col>
+            <Col md="6" className="friend-btn">
+              <Button type="button" onClick={unfriend}>{btnTxt}</Button>
+            </Col>
+          </Row>
+        </CardHeader>
+        <CardBody>
+          <div>
+            <p>{props.username}</p>
+            <p>{props.email}</p>
+          </div>
+        </CardBody>
+      </Card>
+    </Col>
+  );
+}
+
+function state2props(state) {
```

Comment:
I dont think we need to connect every single component since most are just going to be pure components that take in some props from their parents. By connecting every component it does cut down on the piping needed but you break the waterfall of data by linking everything directly to the state

Reply:
I agree, but I just made the default template for the components to connect since it's easier to remove it rather than add it in if we were to need it.

Reply:
+1